mm/mm_gran/mm_grantable.c

```diff
@@ -174,6 +174,7 @@ bool gran_match(const gran_t *gran, size_t posi, size_t size, bool used,
   uint32_t e;   /* expected cell value */
   gatr_t   r;   /* range helper */
 
+  memset(&r, 0, sizeof(r)); /* clear the range explicitly to avoid warnings */
   gran_range(gran, posi, size, &r);
```

Comment:
```suggestion
  if (gran_range(gran, posi, size, &r) < 0)
    {
      memset(&r, 0, sizeof(r));
    }
```

Reply:
Should here return `false` directly, same as what `failure` does?

Reply:
Sure, but some additional testing may be needed. we could memset the `getr_t r` on failure case if we just want to clear the warning, which is consistent with the previous semantics.